File: Programing/HigherOrderFunction/Exercise1/AnswerCSharp/ExecutionUtils.cs
using System;
using System.Diagnostics;

namespace Exercise1
{
    class ExecutionUtils
    {
        public static void ExecuteWithTimeMeasuring(Action function)
	{
	    if (function == null)
	    {
	        throw new ArgumentNullException(nameof(function));
	    }

            var stopWatch = Stopwatch.StartNew();
	    try
	    {
	        function();
	    }
	    finally
	    {
	        Console.WriteLine($"processing time is {stopWatch.ElapsedMilliseconds}ms");
	    }
	}
    }
}

File: Programing/HigherOrderFunction/Exercise1/AnswerJava/ExecutionUtils.java
class ExecutionUtils {
    public static void ExecuteWithTimeMeasuring(Runnable function) {
	if (function == null) {
	    throw new IllegalArgumentException("function is null");
	}
	
	long beginTime = System.currentTimeMillis();
	try {
	    function.run();
	}
	finally {
	    long elapsedTime = System.currentTimeMillis() - beginTime;
	    System.out.println("processing time is " + elapsedTime + "ms");
	}
    }
}
